File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapEnum.java
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation;

import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapEnums;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapFields;

import java.lang.annotation.*;

@Documented
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
@Repeatable(WrapEnums.class)
/**
 * Auto setup Enum
 */
public @interface WrapEnum {
    String mcpName();
    String targetMap();
    String customSrgName() default "none";
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapEnum.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapEnums

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@JvmRepeatable(
    WrapEnums::class
)
/**
 * Auto setup Enum
 */
annotation class WrapEnum(val mcpName: String, val targetMap: MapEnum, val customSrgName: String = "none")